pwiz_tools/Skyline/SkylineGraphs.cs

```diff
@@ -908,6 +908,35 @@ private void charge4MenuItem_Click(object sender, EventArgs e)
             _graphSpectrumSettings.ShowCharge4 = !_graphSpectrumSettings.ShowCharge4;
         }
 
+        private void SynchMzScaleToolStripMenuItemClick(object sender)
+        {
+            if (_graphFullScan == null || !_graphFullScan.Visible
+                                       || _graphSpectrum == null || !_graphSpectrum.Visible)
+                return;
+            Settings.Default.SyncMZScale = synchMzScaleToolStripMenuItem.Checked;

```

Comment:
After this line you should do:
if (!Settings.Default.SyncMZScale)
   return;

It probably won't make a difference, since the graph axes are going to already have been synced with each other, but Skyline should not try to sync them one last time when the user is turning off synchronization.